pkg/statedb/table.go

```diff
@@ -163,10 +163,18 @@ func (t *genTable[Obj]) Get(txn ReadTxn, q Query[Obj]) (Iterator[Obj], <-chan st
 	return &iterator[Obj]{iter}, watchCh
 }
 
-// Insert implements Table
 func (t *genTable[Obj]) Insert(txn WriteTxn, obj Obj) (oldObj Obj, hadOld bool, err error) {
 	var data any
-	data, hadOld, err = txn.getTxn().Insert(t, obj)
+	data, hadOld, err = txn.getTxn().Insert(t, Revision(0), obj)
+	if err == nil && hadOld {
+		oldObj = data.(Obj)
+	}
+	return
+}
+
+func (t *genTable[Obj]) UpdateIfEqual(txn WriteTxn, rev Revision, obj Obj) (oldObj Obj, hadOld bool, err error) {
```

Comment:
Is this `CompareAndSwap`? If so, can we call it the more traditional name?

Reply:
I contemplated that initially, but for some reason didn't go for it. Now `CompareAndSwap` and `CompareAndDelete` seem good, so I'll go with those.

Reply:
Or should we be really explicit and do `CompareRevisionAndSwap`?